File: 백준/Silver/11501. 주식/주식.py
import sys 
input = sys.stdin.readline
sys.setrecursionlimit(10**6)

def find(N, stock):
    stock.reverse()
    max = stock[0]
    sum = 0
    for i in range(1, N):
        if max < stock[i]:
            max = stock[i]
        else:
            sum += max - stock[i]
    return sum    

T = int(input())
for _ in range(T):
    N=int(input())
    stock = list(map(int, input().split()))
    print(find(N, stock))

File: 백준/Silver/11501. 주식/주식.go
package main

import (
	"bufio"
	"fmt"
	"os"
)

var (
	reader = bufio.NewReader(os.Stdin)
	writer = bufio.NewWriter(os.Stdout)
)

func find(n int, stock []int) int {

	max := stock[len(stock)-1]
	sum := 0
	for i := n - 1; i >= 0; i-- {
		if max < stock[i] {
			max = stock[i]
		} else {
			sum += max - stock[i]
		}
	}
	return sum
}
func main() {
	defer writer.Flush()
	var T, n int
	fmt.Fscanf(reader, "%d\n", &T)
	for i := 0; i < T; i++ {
		fmt.Fscanf(reader, "%d\n", &n)
		var stock = make([]int, n)
		for i := 0; i < n; i++ {
			fmt.Fscanf(reader, "%d ", &stock[i])
		}
		fmt.Fprintf(writer, "%d\n", find(n, stock))
	}
}
